### DISCUSSION TITLE
function Install-WinGet {}

### DISCUSSION THREAD

**Discussion by KnifMelti** [Romanitho/Winget-AutoUpdate#108] [Ideas]
Now, in these times of **Winget development** with at lot of previews coming out with better and better builds/releases, why not create a standalone function under the **functions** directory that takes care of updating **Windows Package Manager** at the same time **WAU** itself does an update via **function Update-WAU {}**?
Then, you don't have to reinstall **WAU** completely when the source changes...
...and a check if there's already an "**included_apps.txt**"/"**excluded_apps.txt**" in place when upgrading/installing **WAU** (they could be/are dynamic ones) and if true **-Exclude** them when copying...

**Comment by Romanitho** [comment]
I agree to say that winget development seem to be better and better. But from my point of view, deploying preview version of Winget whould have been something temporary. Winget should be updated via Store. That's why I didn't integrate a process to update Winget...
But in another side, I deploy new version each time they release a new preview one 😅 I can understand it doesn't make sens to you haha. At the end, you're right, we can manage the winget version inside WAU until a real stable version with interesting features is out.

regarding include/exclude apps, if you reinstall WAU you need to give the include/exclude apps with it. I don't want to check if one exists because you may want to force a reinstall. If you update it from sources, it's quite easy to bring your list in it, no ? 😋
Don't know to be honest the work we have to do for the result we can expect
